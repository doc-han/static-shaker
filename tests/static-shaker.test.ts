import staticShaker from "../src/static-shaker";
import * as path from "path";

describe("example01.ts", () => {
  it("start function", () => {
    const output = staticShaker(
      path.join(__dirname, "./files/example01.ts"),
      "start"
    );
    expect(output).toMatchSnapshot();
  });
  it("someOtherFunc", () => {
    const output = staticShaker(
      path.join(__dirname, "./files/example01.ts"),
      "someOtherFunc"
    );
    expect(output).toMatchSnapshot();
  });
});

describe("example02.ts", () => {
  it("getConfig", () => {
    const output = staticShaker(
      path.join(__dirname, "./files/example02.tsx"),
      "getConfig"
    );
    expect(output).toMatchSnapshot();
  });
  it("BoxModal", () => {
    const output = staticShaker(
      path.join(__dirname, "./files/example02.tsx"),
      "BoxModal"
    );
    expect(output).toMatchSnapshot();
  });
});
