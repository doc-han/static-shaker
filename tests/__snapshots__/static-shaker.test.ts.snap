// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example01.ts someOtherFunc 1`] = `
"import path from "path";
import fs from "fs";

const GLOBAL_VAR = "some-global-var";

function someOtherFunc() {
  const fPath = path.join(__dirname, GLOBAL_VAR);
  fs.readFileSync(fPath);
}
"
`;

exports[`example01.ts start function 1`] = `
"import path from "path";

const GLOBAL_VAR = "some-global-var";

function genConfig() {
  return {
    name: GLOBAL_VAR,
    path,
  };
}

const start = () => {
  const config = genConfig();
  console.log(config);
};

export default start;
"
`;

exports[`example02.ts BoxModal 1`] = `
"import React from "react";
import styles from "./googleGPT.modules.css";

const BoxModal: React.FC = () => {
  return (
    (<div className={styles.boxModal}>
      <button>Search with GPT</button>
    </div>)
  );
};
export default BoxModal;
"
`;

exports[`example02.ts getConfig 1`] = `
"const MATCHES = ["*://*.google.com"];

export function getConfig() {
  return {
    matches: MATCHES,
  };
}
"
`;
